source/s3_auto_ranged_get.c

```diff
@@ -691,6 +715,15 @@ static void s_s3_auto_ranged_get_request_finished(
                 auto_ranged_get->synced_data.head_object_completed = true;
                 AWS_LOGF_DEBUG(AWS_LS_S3_META_REQUEST, "id=%p Head object completed.", (void *)meta_request);
                 break;
+            case AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_PART_NUMBER:
+                auto_ranged_get->synced_data.get_first_part_completed = true;
+                AWS_LOGF_DEBUG(AWS_LS_S3_META_REQUEST, "id=%p Get Part Number completed.", (void *)meta_request);
+                if (error_code == AWS_ERROR_S3_GET_PART_SIZE_MISMATCH && found_object_size) {
+                    /* Try to fetch the first part again as a ranged get */
```

Comment:
what happens if AWS_ERROR_S3_GET_PART_SIZE_MISMATCH but we NOT found_object_size?

Reply:
The error should be replaced by why we were not able to find the object size. The only possible case this can happen for now is when Content-Length was present but there was no Content-Range header. So we will report Content-Range not found error to the user.